File: apps/admin_mobile/lib/application/use_cases/clients/get_clients_use_case.dart
import '../../../domain/entities/client_resource.dart';
import '../../../domain/repositories/client_resource_repository.dart';
import '../../../shared/types/paginated_result.dart';

class GetClientsUseCase {
  final ClientResourceRepository _repo;
  GetClientsUseCase(this._repo);

  Future<PaginatedResult<ClientResource>> call({int? page, String? search}) =>
      _repo.getAll(page: page, search: search);
}

File: apps/admin-v2/src/application/use-cases/clients/get-clients.use-case.ts
import type { ClientResourceRepository } from '@/domain/repositories/client-resource.repository';

export class GetClientsUseCase {
  constructor(private repo: ClientResourceRepository) {}

  execute(page?: number, search?: string, withDebt?: boolean) {
    return this.repo.getAll(page, search, withDebt);
  }
}
